src/huggingface_hub/inference/_providers/_common.py

```diff
@@ -157,7 +161,7 @@ def _prepare_base_url(self, api_key: str) -> str:
             logger.info(f"Calling '{self.provider}' provider directly.")
             return self.base_url
 
-    def _prepare_route(self, mapped_model: str) -> str:
+    def _prepare_route(self, mapped_model: str, api_key: Optional[str] = None) -> str:
```

Comment:
```suggestion
    def _prepare_route(self, mapped_model: str, api_key: str) -> str:
```

Let's not make it optional. `_prepare_route` is called at a single place in _common.py and with an `api_key` so we don't benefit from making it optional. 

(to adapt to all providers)

Reply:
true!

Reply:
Addressed in [028396b](https://github.com/huggingface/huggingface_hub/pull/2927/commits/028396bf0c431f4c5abe8afb6b37ee7e8772c0f9)